driver-core/src/main/com/mongodb/client/model/Aggregates.java

```diff
@@ -933,6 +938,90 @@ public static Bson searchMeta(final SearchCollector collector, final SearchOptio
         return new SearchStage("$searchMeta", notNull("collector", collector), notNull("options", options));
     }
 
+    /**
+     * Creates a {@code $vectorSearch} pipeline stage supported by MongoDB Atlas.
+     * You may use the {@code $meta: "vectorSearchScore"} expression, e.g., via {@link Projections#metaVectorSearchScore(String)},
+     * to extract the relevance score assigned to each found document.
+     *
+     * @param queryVector The query vector.
+     * @param path The field to be searched.
+     * @param index The name of the index to use. This index predetermines the number of components of {@code queryVector}.
+     * @param numCandidates The number of candidates.
+     * @param limit The limit on the number of documents produced by the pipeline stage.
+     * @return The {@code $vectorSearch} pipeline stage.
+     *
+     * @mongodb.atlas.manual atlas-vector-search/vector-search-stage/ $vectorSearch
+     * @mongodb.atlas.manual atlas-search/scoring/ Scoring
+     * @mongodb.server.release 7.1
+     * @since 4.11
+     */
+    @Beta(Beta.Reason.SERVER)
+    public static Bson vectorSearch(
+            final FieldSearchPath path,
+            final Iterable<Double> queryVector,
```

Comment:
Should we use Collection? I believe an iterable can be infinite, which does not seem to fit here.

Reply:
To me, the usefulness of restricting the type of `queryVector` from `Iterable` to `Collection` is on par with restricting the type of the argument in `Aggregates.unset` from `List` to `Set` - won't help a user, but may force him to convert a value to a different type before passing it (I don't know why it's a `List`, and not `Collection` or even `Iterable`).


P.S. An assumptions about `Collection` being finite is questionable. When `java.util.stream` was introduced, the authors decided to write _"While collections have a finite size, streams need not." in the documentation of this package. However, I fail to find something in the documentation of `Collection` in support of this. `Collection.size` is documented as _""If this collection contains more than `Integer.MAX_VALUE` elements, returns `Integer.MAX_VALUE`._, which is not inconsistent with the size being infinite.

Reply:
Alright - I see it is also consistent with what we do elsewhere.